next-app/src/routes/projects/[project_code]/meta/types.d.ts

```diff
@@ -0,0 +1,22 @@
+type ProjectDetails = {
+	id: string,
+	code: string,
+	name: string,
+	num_users: number,
+	num_entries: number,
+	num_entries_with_audio: number,
+	num_entries_with_pictures: number,
+	num_unresolved_comments?: number,
+}
+
+type Entry = {
+	senses?: Sense[],
+	[key: string]: Field,
```

Comment:
I don't think this is right. Isn't it just children of lexeme that should be treated like this? Possibly also customFields. Everything else is well defined but it's just getting caught in this catch all. I'd probably define it like this:
```ts
interface Entry {
  senses?: Sense[];
  lexeme: Record<string, Field>;
  customFields: Record<string, Record<string, Field>>;
}
```
I'd probably add all the other fields so we can reuse this type across the whole application but I'm sure you can figure out what that would look like.

Reply:
yeah, I struggled a little with all of this, I'd like to pair on it if you don't mind

Reply:
I've just eliminated all need for explicit typing in here for the time, [fccc305](https://github.com/sillsdev/web-languageforge/pull/1664/commits/fccc3059b003119bdc9e1f5064cc48f0169276d1)